src/synthesizer/components/stellar.py

```diff
@@ -1223,32 +1221,46 @@ def get_line_screen(
             dust_curve_stellar=dust_curve,
         )
 
-    def check_young_old_units(self, young, old):
+    def _check_young_old_units(self, young, old):
         """
         Checks whether the `young` and `old` arguments to many
         spectra generation methods are in the right units (Myr)
 
         Args:
-            young (bool, float):
-                If not False, specifies age in Myr at which to filter
+            young (float/unyt_quantity):
+                If not None, specifies age in Myr at which to filter
                 for young star particles.
-            old (bool, float):
-                If not False, specifies age in Myr at which to filter
+            old (float/unyt_quantity):
+                If not None, specifies age in Myr at which to filter
                 for old star particles.
         """
 
-        if young:
+        if young is not None:
             if isinstance(young, (unyt_quantity)):
                 young = young.to("Myr")
             else:
                 young *= Myr
 
-        if old:
+        if old is not None:
             if isinstance(old, (unyt_quantity)):
                 old = old.to("Myr")
             else:
                 old *= Myr
 
+        # Ensure ages aren't larger than the age of the universe
+        if young is not None and young > 13.8 * Gyr:
```

Comment:
So we are only allowed to use synthesizer for generating spectra in a fiducial Lambda-CDM Universe? This is going to make it difficult to use for, e.g. CAMELS :)

Reply:
You're gonna have to think of a better way, as this kind of hard check isn't going to work. Not sure what you mean by 'magnitudes too old'.

Reply:
That or we enforce units which would be safer

Reply:
For what it's worth: I don't think not having a check here is an option. By having it check for 50 Gyrs as the limit this won't exclude any meaningful cosmology but it will catch any unit issues. We've had the issue this fixes fly under the radar for ages.

Reply:
Ok, I'll enforce units. I prefer that anyway, I thought you would be against it 😂